Formula/p/proto.rb

```diff
@@ -0,0 +1,48 @@
+class Proto < Formula
+  desc "Pluggable multi-language version manager"
+  homepage "https://moonrepo.dev/proto"
+  url "https://github.com/moonrepo/proto/archive/refs/tags/v0.30.0.tar.gz"
+  sha256 "a564e4c8a4d2483b19e02bd1b3cdac5db4ad61cc0e2d217203f0a641897ba7f4"
+  license "MIT"
+  head "https://github.com/moonrepo/proto.git", branch: "master"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "rust" => :build
+
+  on_linux do
+    depends_on "pkg-config" => :build
+  end
+
+  def install
+    system "cargo", "install", *std_cargo_args(path: "crates/cli")
+    generate_completions_from_executable(bin/"proto", "completions", "--shell")
+
+    bin.each_child do |f|
+      basename = f.basename
+
+      (libexec/"bin").install f
+      (bin/basename).write_env_script libexec/"bin"/basename, PROTO_INSTALL_DIR: opt_prefix/"bin"
+    end
+  end
+
+  def caveats
+    <<~EOS
+      To finish the installation, run:
+        proto setup
+    EOS
+  end
+
+  test do
+    system bin/"proto", "install", "node", "19.0.1"
+    node = shell_output("#{bin}/proto bin node").chomp
+    assert_match "19.0.1", shell_output("#{node} --version")
+    path = testpath/"test.js"
```

Comment:
Let's add some whitespace between tests
```suggestion
    assert_match "19.0.1", shell_output("#{node} --version")

    path = testpath/"test.js"
```